tket/src/Circuit/ToffoliBox.cpp

```diff
@@ -355,6 +355,37 @@ void ToffoliBox::generate_circuit() const {
       permute(perm, n_, get_op_ptr(rotation_axis_, 1)));
 }
 
+static bool perm_compare(const state_perm_t &a, const state_perm_t &b) {
+  auto ita = a.begin();
+  auto itb = b.begin();
+  for (; ita != a.end(); ita++) {
```

Comment:
Could refactor this into a static helper function that's called twice with swapped arguments. Up to you.